src/query/interpreter.cpp

```diff
@@ -313,31 +335,76 @@ class ReplQueryHandler final : public query::ReplicationQueryHandler {
-                                                         .ssl = std::nullopt};
-      auto ret = handler_.RegisterReplica(config);
+      const auto [ip, port] = *maybe_ip_and_port;
+      const auto config = replication::ReplicationClientConfig{.name = name,
+                                                               .mode = repl_mode,
+                                                               .ip_address = ip,
+                                                               .port = port,
+                                                               .replica_check_frequency = replica_check_frequency,
+                                                               .ssl = std::nullopt};
+#ifdef MG_ENTERPRISE
+      const auto ret = std::invoke([&config, repl_handler = &this->handler_]() {
+        if (FLAGS_coordinator) {
+          if (!license::global_license_checker.IsEnterpriseValidFast()) {
+            throw QueryException("Trying to use enterprise feature without a valid license.");
+          }
+          return repl_handler->RegisterReplicaOnCoordinator(config);
+        }
+        return repl_handler->RegisterReplica(config);
+      });
+#else
+      const auto ret = handler_.RegisterReplica(config);
+#endif
+
       if (ret.HasError()) {
         throw QueryRuntimeException(fmt::format("Couldn't register replica '{}'!", name));
       }
+
+    } else {
+      throw QueryRuntimeException("Invalid socket address!");
+    }
+  }
+
+#ifdef MG_ENTERPRISE
+  /// @throw QueryRuntimeException if an error ocurred.
+  void RegisterMain(const std::string &socket_address, const std::chrono::seconds main_check_frequency) override {
+    if (!FLAGS_coordinator) {
+      throw QueryRuntimeException("Only coordinator can register main instance!");
+    }
+
+    const auto maybe_ip_and_port =
+        io::network::Endpoint::ParseSocketOrAddress(socket_address, memgraph::replication::kDefaultReplicationPort);
+    if (maybe_ip_and_port) {
+      const auto [ip, port] = *maybe_ip_and_port;
+      const auto config = replication::ReplicationClientConfig{
+          .name = memgraph::replication::kDefaultMainName,
+          .mode = replication::ReplicationMode::ASYNC,  // MAIN is not using actually replication mode, TODO: (andi) Set
+                                                        // it to optional
+          .ip_address = ip,
+          .port = port,
+          .replica_check_frequency = main_check_frequency,  // TODO: (andi) better naming would be check_frequency
+          .ssl = std::nullopt};
+
+      if (const auto ret = handler_.RegisterMainOnCoordinator(config); ret.HasError()) {
+        throw QueryRuntimeException("Couldn't register main!");
+      }
     } else {
       throw QueryRuntimeException("Invalid socket address!");
     }
   }
+#endif
 
   /// @throw QueryRuntimeException if an error occurred.
   void DropReplica(std::string_view replica_name) override {
```

Comment:
Do we need `DROP MAIN/REPLICA` for the coordinator as well?

Reply:
at the moment no, maybe in the future